pkg/core/blockchain.go

```diff
@@ -2530,6 +2530,32 @@ func (bc *Blockchain) verifyAndPoolTx(t *transaction.Transaction, pool *mempool.
 	return nil
 }
 
+// CalculateAttributesFee returns network fee for all transaction attributes that should be
+// paid according to native Policy.
+func (bc *Blockchain) CalculateAttributesFee(tx *transaction.Transaction) int64 {
+	var (
+		feeCache map[transaction.AttrType]int64
```

Comment:
Why this cache? We have one in the `Policy` contract already.

Reply:
Every time we access Policy cache we must call RO lock over it, and with local cache it may be cheaper (lock it once per attribute type, not for every attribute), that's the only reason. Or maybe it's still better to use policy cache, and not to allocate new map on each transaction.

Reply:
Fixed, let's use Policy cache, our GC won't be happy with dozens of tiny maps.